chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -915,15 +891,19 @@ impl<'a, S: BlockchainStorageWrite, V: TransactionVerificationStrategy> Chainsta
     }
 
     fn post_connect_tip(&mut self, tip_index: &BlockIndex) -> Result<(), BlockError> {
+        let tip_height = tip_index.block_height();
         epoch_seal::update_epoch_seal(
             &mut self.db_tx,
             self.chain_config,
-            epoch_seal::BlockStateEvent::Connect(tip_index.block_height()),
+            epoch_seal::BlockStateEvent::Connect(tip_height),
         )?;
+        let tip = self
+            .get_block_from_index(tip_index)?
+            .ok_or(BlockError::BlockAtHeightNotFound(tip_height))?;
```

Comment:
This error is more like an invariant is broken, because we already connected that block so it should exist already. Can you create a new error with something like `InvariantBrokenBlockNotFoundAfterConnect`?

Reply:
Perhaps we should change this to pass the block to both functions when connecting the tip and post-connecting... it's very inefficient to do it this one.

Maybe just throw in a TODO at the connect loop, under the comment "Connect the new chain"